core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1073,7 +1075,26 @@ private void internalPerformTranslogRecovery(boolean skipTranslogRecovery, boole
         } else {
             openMode = EngineConfig.OpenMode.OPEN_INDEX_AND_TRANSLOG;
         }
-        final EngineConfig config = newEngineConfig(openMode, maxUnsafeAutoIdTimestamp);
+
+        boolean assertionsEnabled = false;
+        assert assertionsEnabled = true;
+        // TODO: add this for shrinked indices.
+        if (assertionsEnabled && indexExists) {
```

Comment:
please move this into a method `private boolean assertSomething()` and then do `assert assertSomething();`